speedy-recovery-client/src/service/FhirMappingAdapterSTU2.js

```diff
@@ -0,0 +1,116 @@
+import {
+  getbodySite,
+  getForm,
+  getLocation,
+  getMedName,
+  getName,
+  getPatient,
+  getPhone,
+  getPractitioner,
+  getSeverity,
+  getSummary
+} from "./FhirJsonCommonExtractionTools";
+
+const mapPatientToUserSTU2 = fhirPatientResource => ({
+  //This is a temporary hard-code fix as we have not implemented the searching for a patients' parent
+  role:
+    fhirPatientResource.id === "f0462936-eb4b-4da1-b45a-fbd96ebf8ccb"
+      ? "Parent"
+      : fhirPatientResource.resourceType,
+
+  name: getName(fhirPatientResource.name),
+  birthDate: fhirPatientResource.birthDate
+    ? new Date(fhirPatientResource.birthDate)
+    : "Unknown",
+  gender: fhirPatientResource.gender ? fhirPatientResource.gender : "Unknown",
+  careProvider: "Undefined in STU2",
+  address: getAddress(fhirPatientResource.address),
+  phone: getPhone(fhirPatientResource.telecom),
+  email: "Undefined in STU2"
+});
+
+const mapAppointmentSTU2 = fhirAppResource => ({
+  id: fhirAppResource.id ? fhirAppResource.id : "Unknown",
+  title: getTitle(fhirAppResource.text),
+  status: fhirAppResource.status ? fhirAppResource.status : "Unknown",
+  appType: "Undefined in STU2",
+  indication: "Undefined in STU2",
+  priority: fhirAppResource.priority ? fhirAppResource.priority : "Unknown",
+  description: fhirAppResource.description
+    ? fhirAppResource.description
+    : "Unknown",
+  supportingInfo: "Undefined in STU2",
+  start: fhirAppResource.start ? new Date(fhirAppResource.start) : "Unknown",
+  end: fhirAppResource.end ? new Date(fhirAppResource.end) : "Unknown",
+  created: "Undefined in STU2",
+  comment: fhirAppResource.comment ? fhirAppResource.comment : "Unknown",
+  patient: getPatient(fhirAppResource.participant),
+  practitioner: getPractitioner(fhirAppResource.participant),
+  location: getLocation(fhirAppResource.participant)
+});
+
+const mapConditionSTU2 = fhirCondResource => ({
+  clinicalStatus: "Undefined in STU2",
+  verificationStatus: fhirCondResource.verificationStatus
+    ? fhirCondResource.verificationStatus
+    : "Unknown",
+  severity: getSeverity(fhirCondResource.severity),
+  summary: getSummary(fhirCondResource.code),
+  bodySite: getbodySite(fhirCondResource.bodySite),
+  onsetDateTime: fhirCondResource.onsetDateTime
+    ? new Date(fhirCondResource.onsetDateTime)
+    : "Unknown"
+});
+
+const mapMedicationSTU2 = fhirMedResource => ({
+  id: fhirMedResource.id ? fhirMedResource.id : "Unknown",
+  producer: "Undefined in STU2",
+  name: getMedName(fhirMedResource.code),
+  isBrand:
+    typeof fhirMedResource.isBrand !== "undefined"
+      ? fhirMedResource.isBrand
+      : "Unknown",
+  isOverTheCounter: "Undefined in STU2",
```

Comment:
Is this acutal text we want to show in the user-facing front-end? Would it make sense to set the missing optional fields as `null` and then exclude them from the visualisation in the components? (I think it would be nicer, but not sure if worth the extra work. You decide)

Reply:
Actually, I think it's fine if we leave it as it is. We have more important things to do. ;)